### ISSUE TITLE
warn on setting "when" fact

### ISSUE THREAD

**Issue by pikrzysztof** [ansible/ansible-lint#885]
##### Summary

ansible-lint should warn on this:

```
- set_fact:
    xyz: "foo"
    when: "something"
```

as you probably meant

```
- set_fact:
    xyz: "foo"
  when: "condition"
```

##### Issue Type

- Feature Idea

##### Additional Information

I wasted 5 hours trying to figure out why my playbook does something unexpected. Turns out that indenting your YAMLs with two spaces makes this mistake hard to spot.

**Comment by ssbarnea** [comment]
It should be quite easy to write rule that does that. @pikrzysztof please feel free to make a PR to add it. If you move fast, we may even include it in 4.3

**Comment by webknjaz** [comment]
Also, I'd not special-case just one keyword. If it's a common case, have a list of those.

**Comment by ssbarnea** [comment]
Closing because AFAIK it is not possible to correctly distinguish between a wrong string or a variable. There is nothing the linter can do to help in this case, you could have a variable named condition.
